doc/RELEASE.adoc

```diff
@@ -0,0 +1,120 @@
+- This does not include tests that require building a new node, or integrating with other downstream projects.
+- In some cases we need input from domain experts and/or security audit (e.g., when adding a cryptography builtin). The right group of people should be involved and sign off on the tests and checks.
+5. If no release blocking issue is found, do a formal release and upload the packages to Hackage.
+- If issues are found, create a release branch `release/x.y`, fix the issues on master, backport the fixes to `release/x.y`, tag `x.y.0-rc2`, and go to step 4.
+- Why not just fix the issues on master and tag `x.y.0-rc2` from master?
+It is desirable to minimize the amount of change between `rc1` and `rc2`, because it may reduce the tests and checks that need to be performed against `rc2`.
+For instance, if `plutus-ledger-api` is the only package changed, there is no need to re-run tests on `plutus-core` or `plutus-tx`.
+Another example is if a security audit is done on `rc1`, and the changes in `rc2` does not modify the audited code, then the audit does not need to be re-done.
+
+6. After the release, update version numbers in cabal files to `x.y+1.0`.
+- This may not be the actual version of the next release - it could be `x+1.0.0`. We don't need to be precise here; we just need a version number higher than the latest release.
+
+=== Patch Releases
+
+1. If it is the first time we are making a patch release for version `x.y` (i.e., `x.y.0` -> `x.y.1`), a release branch may not exist for `x.y`. If so, create branch `release/x.y` from the `x.y.0` tag.
+- We create release branches lazily, because we do not expect to make many patch releases.
+2. Backport the needed fixes from master to `release/x.y`.
+3. Run all tests and checks.
+4. If no release blocking issue is found, do a formal release and upload the packages to Hackage.
+- If issues are found, fix them on master, backport the fixes to `release/x.y`, and go to step 3.
+
+== Backwards Compatibility
+
+Plutus releases should meet two backwards compatibility criteria:
+
+1. *Each Plutus release should build with the latest version of Cardano API.*
++
+This makes it possible for Plutus Tools to upgrade to the new Plutus release, without needing a new Cardano API release.
++
+To do so, rather than making changes to the Plutus API that breaks Cardano API (e.g., changing the type of a function), we should temporarily keep both the old Plutus API and the new Plutus API, until a new Cardano API version is released that no longer depends on the old Plutus API.
++
+This requirement will likely go away once Cardano API starts to make more frequent releases.
+
+2. *Plutus language behaviors should be preserved.*
++
+When we start developing a new Plutus language version (e.g., Plutus V3), initially the language version is considered "unreleased", and should be guarded behind a flag to make it unusable.
```

Comment:
```suggestion
When we start developing a new Plutus ledger language version (e.g., Plutus V3), initially the language version is considered "unreleased", and should be guarded behind a flag to make it unusable.
```

Reply:
Interesting, we should work out how we're going to do this technically. Since I think we _do_ want to test this, and we maybe even want the ledger to be able to start putting together the scaffolding for the new version also. So if anything I think the flag should be in the _ledger_.